### DISCUSSION TITLE
where is tineye

### DISCUSSION THREAD

**Discussion by mehr32** [searxng/searxng#2607] [Q&A] [answered]
Hello
I have forked a version of searxng and am using it from my personal computer. I was reading the code of the project when I got no particular tineye but I couldn't find a place in searxng to use it at the time of use. What is the problem?

**Comment by return42** [comment, marked as answer]
Tineye is a reverse image search service .. [SearXNG engine only supports ‘searching by URL’ and it does not use the official API](https://docs.searxng.org/dev/engines/online_url_search/tineye.html#tineye)

Tineye is disabled by default since it response is very slow (timoute set to 9sec). You can activate the engine in the preferences, but I would recommend to use the [search-syntax](https://docs.searxng.org/user/search-syntax.html) and select the engine with its shortcut `!tin` .. by example if you have a URL of an image like 

- https://static.vecteezy.com/system/resources/previews/003/759/921/original/alarm-clock-cartoon-isolated-free-vector.jpg 

you can start a reverse search by:

- [`!tin https://static.vecteezy.com/system/resources/previews/003/759/921/original/alarm-clock-cartoon-isolated-free-vector.jpg`](https://darmarit.org/searx/search?q=%21tin+https%3A%2F%2Fstatic.vecteezy.com%2Fsystem%2Fresources%2Fpreviews%2F003%2F759%2F921%2Foriginal%2Falarm-clock-cartoon-isolated-free-vector.jpg)

---

Next time please do not open a bug report for a question you have / I moved your ticket into the Q&A discussion.
